packages/ui/src/components/inputs/number-input.tsx

```diff
@@ -103,6 +115,7 @@ export const NumberInput = forwardRef<HTMLInputElement, NumberInputProps>(
             hideStepper ? null : (
               <div className="flex flex-col">
                 <Button
+                  type="button"
```

Comment:
Any `button` placed inside a form without `type` attribute acts as a submit button. Hence it is needed.